src/Nncase.Passes/Rules/Neutral/MatMulToConv2D.cs

```diff
@@ -107,6 +109,92 @@ public sealed partial class BroadcastMatMulToConv2D : IRewriteRule
     }
 }
 
+/// <summary>
+/// Transform broadcast <see cref="IR.Math.MatMul"/> b to <see cref="IR.Math.MatMul"/> a and squeeze matmul to 3D.
+/// </summary>
+[RuleGenerator]
+public sealed partial class BroadcastMatMul : IRewriteRule
+{
+    /// <inheritdoc/>
+    public IPattern Pattern { get; } =
+        IsMatMul(
+            "matMul",
+            "matMulCall",
+            _ => true,
+            IsWildcard("a") with { TypePattern = HasRank(r => r > 2, "Rank > 2") },
```

Comment:
add has_fixed_shape to avoid error of dynamic shape.

Reply:
Done